### PR TITLE
Thread tuning

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary

* Not spawning threads when the parent thread blocks for completion of the child but instead just run it in the parent - was the case for getting validation packages and running validation
* Not using threads from the thread pool for timeouts as this just binds a thread to have it sleep for a minute => this would eat up the thread pool quickly and made zome calls lock up, waiting for a new thread to become available

## testing/benchmarking notes

This makes stress tests much faster and solves the zome call lock-up issue.

## followups

We should implement timeouts with just one thread that polls the state looking for timeouts to trigger instead of creating a thread per task just to have it sleep.

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

**Comment by willemolding** [comment]
@freesig does this touch on any of the things you were seeing in the profiler?

**Comment by lucksus** [comment]
@zippy, @willemolding, please note I've added another commit that removes what has become dead code by making validation and validation package calls synchronous and not mediated by actions and the State.

File: crates/core/src/nucleus/actions/build_validation_package.rs [hunk 1]
```diff
@@ -9,25 +8,20 @@ use crate::{
...
     error::HolochainError,
     signature::Provenance,
-    ugly::lax_send_sync,
     validation::{ValidationPackage, ValidationPackageDefinition::*},
 };
-use snowflake;
-use std::{pin::Pin, sync::Arc, vec::Vec};
+use std::{sync::Arc, vec::Vec};
 
 pub async fn build_validation_package<'a>(
```
Comment (willemolding) [inline crates/core/src/nucleus/actions/build_validation_package.rs]: I believe you can drop the `async` from this function now right?

File: crates/core/src/nucleus/actions/run_validation_callback.rs [hunk 2]
```diff
@@ -21,95 +18,45 @@ use holochain_metrics::Metric;
 /// Dispatches an `Action::ReturnValidationResult` after completion of the WASM call.
 /// Returns a future that waits for the result to appear in the nucleus state.
 pub async fn run_validation_callback(
```
Comment (willemolding) [inline crates/core/src/nucleus/actions/run_validation_callback.rs]: and same here?

**Comment by freesig** [comment]
> @freesig does this touch on any of the things you were seeing in the profiler?

It's a little hard to tell as my knowledge of how the whole code base plays together is very limited but this isn't the stuff I was looking at in our call. 

I can profile this PR if you like though

**Comment by freesig** [comment]
This is looking faster but still getting this pattern 
image

**Pull request merged**